offchain/rollups-events/src/main.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright Cartesi Pte. Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+use std::time::Duration;
+
+use backoff::ExponentialBackoffBuilder;
+use redacted::{RedactedUrl, Url};
+use rollups_events::{Broker, BrokerConfig, BrokerStream};
+use serde::{Deserialize, Serialize};
+
+const REDIS_TLS_PORT: &str = "6379";
+const STREAM_KEY: &'static str = "test-stream";
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct MockPayload {
+    data: String,
+}
+
+struct MockStream {}
+
+impl BrokerStream for MockStream {
+    type Payload = MockPayload;
+
+    fn key(&self) -> &str {
+        STREAM_KEY
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let config = create_config();
+    let mut broker =
+        Broker::new(config).await.expect("Failed to create Broker");
+
+    let stream = MockStream {};
+    let payload = MockPayload {
+        data: "test-data".into(),
+    };
+    let id = broker
+        .produce(&stream, payload.clone())
+        .await
+        .expect("Failed to produce an event");
+
+    let event = broker
+        .peek_latest(&stream)
+        .await
+        .expect("Failed to peek at latest event")
+        .expect("No event found in the stream");
+
+    assert_eq!(event.id, id);
+    assert_eq!(event.payload, payload);
+}
+
+fn create_config() -> BrokerConfig {
+    let redis_endpoint =
+        Url::parse(&format!("rediss://localhost:{}", REDIS_TLS_PORT))
```

Comment:
It's not common to see `rediss:// ` but just `redis://`, and with `redis-cli`  you just add `-tls` to the CLI argument.

Some language examples:

**python**
```python
  r = redis.StrictRedis(
    host='localhost',
    port=12000,
    ssl=True,
    ...
```

**javascript**
```javascript
var client = redis.createClient(12000, '127.0.0.1', {
  password: 'secretdb01',
  tls: ssl,
});
```

---

If we  adopt `rediss://`,  I'd be VERY clear about it on the documentation.

Reply:
As far as I could gather from the documentation and from the [code](https://github.com/redis-rs/redis-rs/blob/main/redis/src/connection.rs#L207), using the `rediss` scheme is the only way to create a TLS connection in the rust-rs crate. 

I'll add the skeleton README file with this info.